File: date/3.14/206.cpp
class Solution
{
public:
    ListNode *reverseList(ListNode *head)
    {
        if (head == NULL)
        {
            return head;
        }

        if (head->next == NULL)
        {
            return head;
        }

        stack<ListNode *> nodeStack;
        for (auto i = head; i != nullptr; i = i->next)
        {
            nodeStack.push(i);
        }
        auto ans = nodeStack.top();
        auto currentNode = nodeStack.top();
        nodeStack.pop();
        while (!nodeStack.empty())
        {
            auto current = nodeStack.top();
            nodeStack.pop();
            currentNode->next = current;
            currentNode = current;
            current->next = NULL;
        }
        return ans;
    }
};

File: date/3.14/206.py
class Solution:
    def reverseList(self, head):
        cur=head
        nodeStack=[]
        while cur!=None:
            nodeStack.append(cur)
            cur=cur.next
        
        ans=nodeStack.pop()
        cur=ans
        while nodeStack.count>0:
            n=nodeStack.pop()
            n.next=None
            cur.next=n
            cur=n

        return ans